pkg/common/templates/k8sresource_funcs.go

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2020 Red Hat, Inc.
+// Copyright Contributors to the Open Cluster Management project
+
+package templates
+
+import (
+  metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+  "github.com/golang/glog"
+  base64 "encoding/base64"
+)
+
+
+// retrieves value of the key in the given Secret, namespace
+func fromSecret(namespace string, secretname string,  key string) (string , error) {
+  glog.V(2).Infof("fromSecret for namespace: %v, secretname: %v, key:%v", namespace, secretname, key)
+
+  secretsClient := (*kubeClient).CoreV1().Secrets(namespace)
+  secret, getErr := secretsClient.Get(secretname, metav1.GetOptions{})
+
+  if getErr != nil {
+    glog.Errorf("Error Getting secret:  %v", getErr)
+    return "" , getErr
+  }
+
+  keyVal := secret.Data[key]
+  glog.V(2).Infof("Secret Key:%v, Value: %v", key, keyVal)
+
+  // when using corev1 secret api, the data is returned decoded ,
+  // re-encododing to be able to use it in the referencing secret
+  sEnc := base64.StdEncoding.EncodeToString(keyVal)
+  glog.V(2).Infof("encoded secret Key:%v, Value: %v", key, sEnc)
+
+  return sEnc, nil
+}
+
+// retrieves value for the key in the given Configmap, namespace
+func fromConfigMap(namespace string, cmapname string,  key string) (string , error) {
+  glog.V(2).Infof("fromConfigMap for namespace: %v, configmap name: %v, key:%v", namespace, cmapname, key)
+
+  configmapsClient := (*kubeClient).CoreV1().ConfigMaps(namespace)
+  configmap, getErr := configmapsClient.Get(cmapname, metav1.GetOptions{})
+
+  if getErr != nil {
+      glog.Errorf("Error getting configmap:  %v", getErr)
+    return "" , getErr
+  }
+  //glog.Errorf("Configmap is %v", configmap)
```

Comment:
I think this can be cleaned up too

Reply:
by default Info is not logged and it is also at level 2. .V(2).Infof . So this would appear in the logs only if log level is elevated to level 2 and Info is enabled